### PR TITLE
Fix issue when logout is called several times for the same server

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
When we call logout several times (mainly when we enter the app and several "logout if needed" functions are executed at the same time) we had an issue with the theme being undefined in the server screen.

This makes so only one onlogout process can be running for each server at the same time.

#### Ticket Link
FIX https://mattermost.atlassian.net/browse/MM-54710
FIX https://github.com/mattermost/mattermost-mobile/issues/7577

#### Release Note
```release-note
Fix "Cannot read property 'centerChannelColor' of undefined" issue
```

### CODE REVIEW

File: app/managers/session_manager.ts [hunk 1]
```diff
@@ -190,10 +192,11 @@ class SessionManager {
 
             relaunchApp({launchType, serverUrl, displayName});
         }
+        this.terminatingSessionUrl.delete(serverUrl);
```
Comment (enahum) [inline app/managers/session_manager.ts:195]: Maybe a setTimeout here could also help?

**Review by enahum — APPROVED** [review]

**Review by harshilsharma63 — APPROVED** [review]

File: app/managers/session_manager.ts (hunk 1 above)
Comment (larkox) [reply app/managers/session_manager.ts:195]: Not sure how that will help. I don't expect anything inside this function to crash or take an infinite amount of time, so... a timeout would only cause problems if things get executed out of place. But 0/5.

**Review by yasserfaraazkhan — APPROVED** [review]
Tested this on android and iOS 17 by generating notifications and opening them when app is in background, opening from notification tray. The app didn't show unexpected error crash.

**Comment by enahum** [comment]
> Tested this on android and iOS 17 by generating notifications and opening them when app is in background, opening from notification tray. The app didn't show unexpected error crash.

Isnt this about logging out multiple times?

**Comment by larkox** [comment]
@yasserfaraazkhan Can you test the following scenario?
- Have the app closed
- Receive a notification
- From the browser, invalidate the mobile session
- Follow the notification

It should send you to the login screen with some message regarding being logged out, and it should not crash.

**Comment by yasserfaraazkhan** [comment]
@larkox I've followed the same qa steps and opening notification after app is logged out, it takes use to login screen.

rn_image_picker_lib_temp_0ae10b44-0dab-4d07-8269-6844cfae5b24

**Review by yasserfaraazkhan — APPROVED** [review]

**Pull request merged**